### PR TITLE
Update Random signatures

### PR DESCRIPTION
**Pull request by snaka**
There are a few things I'm not sure about the changes, so it would be helpful if you could check the comments.

### CODE REVIEW

File: core/random.rbs [hunk 1]
```diff
@@ -104,14 +31,16 @@ class Random < Object
...
-  # Alias of Random::DEFAULT.rand.
-  #
   def self.rand: () -> Float
-               | (Integer | ::Range[Integer] max) -> Integer
-               | (Float | ::Range[Float] max) -> Float
+               | (Integer max) -> Integer
+               | (Float max) -> Float
+               | (::Range[Integer] range) -> Integer?
+               | (::Range[Float] range) -> Float?
+
```
Comment (snaka) [inline core/random.rbs:39]: The original had a definition for `Numeric`, but I am not sure if it is needed.

File: core/random.rbs [hunk 2]
```diff
@@ -129,141 +58,53 @@ class Random < Object
   #     [ rand, rand ]           # => [0.1915194503788923, 0.6221087710398319]
   #
   def self.srand: (?Integer number) -> Integer
-end
-
-# The default Pseudorandom number generator.  Used by class methods of Random.
-#
-#
-Random::DEFAULT: Random
```
Comment (snaka) [inline core/random.rbs:137]: I ran `rbs prototype runtime --merge Random` and the `Random::DEFAULT` constant was removed from the definition file.

Do I need to specify any options when I run `rbs prototype`?

File: core/random.rbs [hunk 3]
```diff
@@ -129,141 +58,53 @@ class Random < Object
...
-  #     SecureRandom.uuid #=> "62936e70-1815-439b-bf89-8492855a7e6b"
-  #
-  # The version 4 UUID is purely random (except the version). It doesn't contain
-  # meaningful information such as MAC addresses, timestamps, etc.
-  #
-  # The result contains 122 random bits (15.25 random bytes).
-  #
-  # See RFC 4122 for details of UUID.
-  #
-  def uuid: () -> String
```
Comment (snaka) [inline core/random.rbs:268]: When I run `rbs prototype runtime --merge Random`, the `Random::Formatter` module is removed from the definition file.

Do I need to specify any options when I run `rbs prototype`?

Comment (snaka) [reply core/random.rbs:268]: Removing `Random::Formatter` seems to be causing the test to fail...

File: core/random.rbs (hunk 1 above)
Comment (ksss) [reply core/random.rbs:39]: In what cases does it return `nil` in `Random.rand`?
The last check seems to have eliminated the possibility of returning nil or false.

https://github.com/ruby/ruby/blob/342e7a094a70d6f90b96262c88177dae32976c85/random.c#L1388-L1389

There is no such description in ruby/spec either.

https://github.com/ruby/spec/blob/a8afb240adcb413994d4de4d33f3f725105811da/core/random/rand_spec.rb

**Comment by snaka** [comment]
Oh, I found that the pull request already existed.
https://github.com/ruby/rbs/pull/470

This pull request will be taken down.
(I'll put the URL of the above PR in the row of the `Random` class in Spreadsheet)

File: core/random.rbs (hunk 1 above)
Comment (snaka) [reply core/random.rbs:39]: Thank you for your comment and for the helpful information.

As for the part you asked about, I referred to the following document.

https://docs.ruby-lang.org/ja/3.0/method/Random/i/rand.html

I have read the above and confirmed that there is a difference between RDoc and the description.
However, I was not able to follow the C implementation.

So, I wrote the RBS according to the above document for the time being.

**Pull request closed without merging**